src/Features/Protocol/LanguageServices/Extensions/RoslynToProtocolExtensions.cs

```diff
@@ -0,0 +1,101 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.CodeAnalysis.Text;
+using Microsoft.VisualStudio.LanguageServer.Protocol;
+using VSLocation = Microsoft.VisualStudio.LanguageServer.Protocol.Location;
+using VSSymbolKind = Microsoft.VisualStudio.LanguageServer.Protocol.SymbolKind;
+
+namespace Microsoft.CodeAnalysis.Protocol.LanguageServices.Extensions
+{
+    internal static class RoslynToProtocolExtensions
+    {
+        public static Uri ToUri(this Document document)
+        {
+            return new Uri(document.FilePath);
+        }
+
+        public static Position ToPosition(this LinePosition linePosition)
+        {
+            return new Position { Line = linePosition.Line, Character = linePosition.Character };
+        }
+
+        public static Range ToRange(this LinePositionSpan linePositionSpan)
+        {
+            return new Range { Start = linePositionSpan.Start.ToPosition(), End = linePositionSpan.End.ToPosition() };
+        }
+
+        public static Range ToRange(this TextSpan textSpan, SourceText text)
+        {
+            var linePosSpan = text.Lines.GetLinePositionSpan(textSpan);
+            return linePosSpan.ToRange();
+        }
```

Comment:
I don't like these conversion methods as extensions. This will get super confusing, especially since the LSP type names collide with System and Roslyn types.

I'd prefer to have `ProtocolConversions` static class where all these will be regular static methods.